static/app/views/detectors/components/detectorListTable/index.tsx

```diff
@@ -71,44 +78,146 @@ function DetectorListTable({
+    (id: string) => {
+      const newSelected = new Set(selected);
+      if (newSelected.has(id)) {
+        newSelected.delete(id);
+      } else {
+        newSelected.add(id);
+      }
+      setSelected(newSelected);
+    },
+    [selected]
+  );
+
+  const canEnable = useMemo(
+    () => detectors.some(d => selected.has(d.id) && !d.enabled),
+    [detectors, selected]
+  );
+  const canDisable = useMemo(
+    () => detectors.some(d => selected.has(d.id) && d.enabled),
+    [detectors, selected]
+  );
+
+  const canEditDetectors = useMemo(() => {
+    const selectedDetectors = detectors.filter(d => selected.has(d.id));
+    const projectToDetectors: Record<string, Detector[]> = {};
+
+    if (
+      organization.access.includes('org:write') ||
+      organization.access.includes('org:admin')
+    ) {
+      return true;
+    }
+
+    for (const detector of selectedDetectors) {
+      const projectId = detector.projectId;
+      if (!projectToDetectors[projectId]) {
+        projectToDetectors[projectId] = [];
+      }
+      projectToDetectors[projectId]?.push(detector);
+    }
+
+    for (const projectId of Object.keys(projectToDetectors)) {
+      const project = projects.find(p => p.id === projectId) ?? undefined;
+      if (!project) {
+        return false;
+      }
+
+      if (hasEveryAccess(['alerts:write', 'project:read'], {organization, project})) {
+        // team admins can modify all detectors for projects they have admin access to
+        if (project.access.includes('project:write')) {
+          continue;
+        }
+        // members can modify only user-createable detectors for projects they have access to
+        if (
+          projectToDetectors[projectId]?.every(d => detectorTypeIsUserCreateable(d.type))
+        ) {
+          continue;
+        }
+      }
+      return false;
+    }
+
+    return true;
+  }, [selected, detectors, organization, projects]);
+
   return (
     <Container>
       <DetectorListSimpleTable>
-        <SimpleTable.Header>
-          <HeaderCell sortKey="name" sort={sort}>
-            {t('Name')}
-          </HeaderCell>
-          <HeaderCell data-column-name="type" divider sortKey="type" sort={sort}>
-            {t('Type')}
-          </HeaderCell>
-          <HeaderCell
-            data-column-name="last-issue"
-            divider
-            sortKey="latestGroup"
-            sort={sort}
-          >
-            {t('Last Issue')}
-          </HeaderCell>
-          <HeaderCell data-column-name="assignee" divider sort={sort}>
-            {t('Assignee')}
-          </HeaderCell>
-          <HeaderCell
-            data-column-name="connected-automations"
-            divider
-            sortKey="connectedWorkflows"
-            sort={sort}
-          >
-            {t('Automations')}
-          </HeaderCell>
-        </SimpleTable.Header>
+        {selected.size === 0 ? (
+          <SimpleTable.Header>
+            <HeaderCell sortKey="name" sort={sort}>
```

Comment:
I think we need there to be a select all checkbox here even when there is not a selection

Reply:
update: decided we'll make this update to the monitor and automation headers in a future PR